### DISCUSSION TITLE
Using CNs which include special characters

### DISCUSSION THREAD

**Discussion by danfoley1991** [strimzi/strimzi-kafka-operator#9116] [Q&A] [answered]
I am using strimzi with external user certificates so I am not using the user operator for generation. Authentication type is set to `tls-external` to enable this.
The certificates that are external to me have the CN defined in this format: `J Bloggs (sbloggs)`
As you can see, they contain spaces, brackets which are not allowed in kubernetes names. When creating kafka users the metadata.name is used to construct the expected CN. 
Am I misunderstanding the documentation and this functionality is available to me?
If it is not, is there any workarounds that you can think of?

The certificate CNs are fixed.

**Comment by scholzj** [comment, marked as answer]
There is no way to use a KafkaUser resource with a certificate like that. The CN needs to be a valid Kubernetes resource name. So you either have to fix the certificate or you can disable the User Operator and manage the ACLs / Quotas for which you likely use the KafkaUser resource your self using the Kafka Admin API / Kafka CLI tools.

**Reply by danfoley1991** [reply]
Thanks for the quick reply Jakub! I'll go another route to workaround the certificates.
